src/game/q_shared.c

```diff
@@ -656,19 +656,32 @@ void QDECL Com_sprintf(char *dest, int size, const char *fmt, ...) {
 }
 
 // Nico, secured version from etlegacy
-char *QDECL va(char *format, ...) {
-	va_list     argptr;
-	static char string[2][32000]; // in case va is called by nested functions
-	static int  index = 0;
-	char        *buf;
-
-	buf = string[index & 1];
-	index++;
+// suburb, updated to latest etlegacy version, fixes nested calls
+char *QDECL va(const char *format, ...) {
+	va_list       argptr;
+	static char   temp_buffer[MAX_VA_STRING];
+	static char   string[MAX_VA_STRING];  // in case va is called by nested functions
+	static size_t index = 0;
+	char          *buf;
+	size_t        len;
 
 	va_start(argptr, format);
-	Q_vsnprintf(buf, sizeof (*string), format, argptr);
+	vsprintf(temp_buffer, format, argptr); // Q_vsnprintf ???
```

Comment:
this should be secured:
`Q_vsnprintf(temp_buffer, sizeof (temp_buffer), format, argptr);`